mmv1/products/cloudidentity/terraform.yaml

```diff
@@ -77,9 +77,14 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         default_from_api: true
       preferredMemberKey: !ruby/object:Overrides::Terraform::PropertyOverride
         default_from_api: true
+      roles: !ruby/object:Overrides::Terraform::PropertyOverride
+        is_set: true
+        set_hash_func: cloudIdentityGroupMembershipRoleHash
```

Comment:
Why do we need a custom hash func here? The default one should do the same.

Reply:
I was expecting a new field might be added in the future, and hashing should work correctly after that. The API constraint is:

> Must not contain duplicate MembershipRoles with the same name.

Actually beta version of API has an extra field `expiryDetail`.

https://cloud.google.com/identity/docs/reference/rest/v1beta1/groups.memberships#MembershipRole

Reply:
Ah- this is one of provider development's gotchas. The hash function doesn't do _quite_ what most users expect it to. It doesn't identify unique elements in the set- instead, the hash is used to perform the _entire diff for the entry_ in a little similar capacity to a diff suppress func.

Using this setfunc when looking at the old and new state for a set, Terraform would hash all of the entries based on their name alone, and would treat any entries with matching names as identical even if their `expiryDetail` was different. It doesn't match entries to each other *and then* compare, it just matches entries.

We can safely remove this, and use the default- it'll do the right thing.

Reply:
> Using this setfunc (...snip...) any entries with matching names as identical even if their `expiryDetail` was different.

Oh I see, that rings a bell. Removed dca9ddb3059ffe60b9099d9ace3c52ad2c996c1b